### PR TITLE
Convert integration tests into unit tests

### PR DESCRIPTION
**Pull request by WillNilges**
This PR copies the integration tests we had and turns them into unit tests, tweaking a few things and gating the integration tests behind an env var as well.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -48,15 +48,29 @@ We use `playwright` to do integration tests. You can run them with the following
...
 
+## Tips for Testing
+
+### UI Mode
+
+Run with `npx playwright test --ui` to get a nice UI to see your tests with.
+
+https://playwright.dev/docs/running-tests#run-tests-in-ui-mode
+
+### Headed Mode
```
Comment (Andrew-Dickinson) [inline README.md:59]: What's the difference between UI and Headed?

File: app/api.ts [hunk 2]
```diff
@@ -1,4 +1,5 @@
-import { JoinFormInput, JoinFormResponse, NNAssignFormInput, NNAssignFormResponse, QueryFormInput, QueryFormResponse } from "@/app/io";
+"use client"
```
Comment (Andrew-Dickinson) [inline app/api.ts:1]: Is this a typo?

File: tests/join_form.spec.ts [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
-// Integration tests for the Join Form.
+// Unit tests for the Join Form.
 //
-// These tests can hit either a self-hosted dev instance of MeshDB
-// (See https://github.com/nycmeshnet/meshdb) or it can hit the beta
-// env.
+// These tests will mock a connection to MeshDB. It is simply making sure that
+// the form creates a good-looking payload and can hit a mock API.
 
 // TODO (wdn): I want to test
```
Comment (Andrew-Dickinson) [inline tests/join_form.spec.ts:14]: Still remaining?

**Review by Andrew-Dickinson — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (WillNilges) [reply README.md:59]: Headed shows a browser window, and quits when its done. It's more for just seeing _exactly_ what's happening if you need to. UI shows a custom Playwright UI with a few sidebars and tools that make it easier to interact with.

File: app/api.ts (hunk 2 above)
Comment (WillNilges) [reply app/api.ts:1]: Nope.

Comment (Andrew-Dickinson) [reply app/api.ts:1]: What does it do?

File: tests/join_form.spec.ts (hunk 3 above)
Comment (WillNilges) [reply tests/join_form.spec.ts:14]: eventually™

File: app/api.ts (hunk 2 above)
Comment (WillNilges) [reply app/api.ts:1]: forces the API calls to happen on the client. I had some trouble where it was behaving as if it was happening serverside, and that made it impossible to test (I spent literally all of saturday trying to make it work and got nowhere) so I gave up and went to this instead. Currently, the expected behavior is that the client is calling meshdb anyway.

**Pull request merged**